doc/source/howtos/crash-recovery.rst

```diff
@@ -0,0 +1,295 @@
+which controls the status and consistency of each node as well as the status
+of the whole cluster. This allows maintaining the data integrity more efficiently than
+with traditional asynchronous replication without losing safe writes
+on multiple nodes at the same time.
+
+However, there scenarios where the database service can stop with no node being
+able to serve requests.
+
+.. contents::
+   :local:
+
+.. _pxc-crash-recovery-scenario-1:
+
+Scenario: Node A is gracefully stopped
+================================================================================
+
+In a three node cluster (node A, Node B, node C), one node (node A, for example)
+is gracefully stopped: for the purpose of maintenance, configuration change,
+etc.
+
+In this case, the other nodes receive a "good bye" message from the stopped node
+and the cluster size is reduced; some properties like quorum calculation or auto
+increment are automatically changed. As soon as node A is started again, it
+joins the cluster based on its :variable:`wsrep_cluster_address` variable in
+:file:`my.cnf`.
+
+If the the writeset cache (:variable:`gcache.size`) on nodes B and/or C still
+has all the transactions executed while node A was down, joining is possible via
+:term:`IST`. If :term:`IST` is impossible due to missing transactions in donor’s
+gcache, the fallback decision is made by the donor and :term:`SST` is started
+automatically.
+
+.. _pxc-crash-recovery-scenario-2:
+
+Scenario: Two nodes are gracefully stopped
+================================================================================
+
+Similar to :ref:`pxc-crash-recovery-scenario-1`, the cluster size is reduced to
+1 --- even the single remaining node C forms the primary component and is able to
+serve client requests. To get the nodes back into the cluster, you just need
+to start them.
+
+However, node C will be switched to “Donor/Desynced” state as it has to provide
+the state transfer to at least first joining node. It is still possible to
+read/write to it during that process, but it may be much slower, which depends
+on how large amount of data should be sent during the state transfer. Also, some
+load balancers may consider the donor node as not operational and remove it from
+the pool. So, it is best to avoid the situation when only one node is up.
+
+If you restart node A and then node B, you may want to make sure note B does not
+use node A as the state transfer donor: node A may not have all the needed
+writesets in its gcache. Specify node C node as the donor in your configuration
+file and start the `mysql` service:
+
+.. code-block:: bash
+
+   $ service mysql start --wsrep_sst_donor=nodeC
+
+.. _pxc-crash-recovery-scenario-3:
+
+Scenario: All three nodes are gracefully stopped
+================================================================================
+
+The cluster is deformed and the problem is to initialize it again. It is
```

Comment:
Maybe it's better to say "completely stopped" or shutdown rather than "deformed".